### ISSUE TITLE
Migrate to CWL workflow definition

### ISSUE THREAD

**Issue by matbun** [interTwin-eu/T6.5-AI-and-ML#11]
Currently a digital twin workflow is defined using YAML files witha custom format. Migrate to the standard [CWL](https://www.commonwl.org/) workflow composition and engine.

**Comment by matbun** [comment]
I tried to run the MNIST workflow with:

```bash
conda run -p ./.venv python run-workflow.py -f ./use-cases/mnist/training-workflow.yml --cwl
``` 

but it breaks, because the file `./use-cases/mnist/pl-config.yml` is not added to CWL tmp context.

Error stack:

```text
...
FileNotFoundError: [Errno 2] No such file or directory: 
'./use-cases/mnist/pl-config.yml'

ERROR conda.cli.main_run:execute(47): `conda run itwinai train --config /tmp/mbunino/i7z0yak5/stgfca12da7-5d52-470b-9ef0-e3fe49d80811/mnist-ai.yml --input /tmp/mbunino/i7z0yak5/stg9561b039-4d7f-4c32-8afa-70b719c30b9e/preproc-images-train` failed. (See above for error)
WARNING [job training] exited with status: 1
WARNING [job training] completed permanentFail
WARNING [step training] completed permanentFail
INFO [workflow ] completed permanentFail
WARNING Final process status is permanentFail
Traceback (most recent call last):
  File "/afs/cern.ch/work/m/mbunino/private/itwin/T6.5-AI-and-ML/run-workflow.py", line 69, in <module>
    subprocess.run(
  File "/afs/cern.ch/work/m/mbunino/private/itwin/T6.5-AI-and-ML/.venv/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'cwltool use-cases/mnist/workflow.cwl ./use-cases/mnist/training-workflow.yml' returned non-zero exit status 1.

ERROR conda.cli.main_run:execute(47): `conda run python run-workflow.py -f ./use-cases/mnist/training-workflow.yml --cwl` failed. (See above for error)
```

May I ask to fix this issue, so that I can complete the MNIST use case?

**Comment by matbun** [comment]
Moving to next milestone, to avoid blocking MVP release.

For next release, need to fix also #28

**Comment by matbun** [comment]
CWL fails for some reason... 
To reproduce:
```bash
micromamba run -p ./.venv python run-workflow.py -f ./use-cases/mnist/training-workflow.yml --cwl
```

```text
Traceback (most recent call last):
  File "/home/mbunino/itwin/test/run-workflow.py", line 125, in <module>
    subprocess.run(
  File "/home/mbunino/itwin/test/.venv/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'cwltool --leave-tmpdir --outdir=./use-cases/mnist/data use-cases/mnist/workflow.cwl ./use-cases/mnist/training-workflow.yml' returned non-zero exit status 1.

```
